source/server-discovery-and-monitoring/tests/unified/logging-sharded.yml

```diff
@@ -0,0 +1,256 @@
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+          - client:
+              id: *client
+              observeLogMessages:
+                topology: debug
+              observeEvents:
+              - serverHeartbeatSucceededEvent
+              uriOptions:
+                heartbeatFrequencyMS: 500
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: *client
+          event:
+            serverHeartbeatSucceededEvent: {}
+          count: 3
+    expectLogMessages:
+      - client: *client
+        ignoreExtraLogs: true
+        ignoreLogMessages:
+          - data:
+              message: "Server heartbeat started"
+          - data:
+              message: "Starting server monitoring"
+          - data:
+              message: "Stopped server monitoring"
+          - data:
+              message: "Topology description changed"
+        messages:
+          - level: debug
+            component: topology
+            data:
+              message: "Starting topology monitoring"
+              topologyId: { $$exists: true }
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+              awaited: { $$exists: true }
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+              driverConnectionId: { $$exists: true }
+              serverConnectionId: { $$exists: true }
+              durationMS: { $$type: [int, long] }
+              reply:
+                $$matchAsDocument:
+                  "$$matchAsRoot":
+                    ok: 1
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+              awaited: { $$exists: true }
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+              driverConnectionId: { $$exists: true }
+              serverConnectionId: { $$exists: true }
+              durationMS: { $$type: [int, long] }
+              reply:
+                $$matchAsDocument:
+                  "$$matchAsRoot":
+                    ok: 1
+          - level: debug
+            component: topology
+            data:
+              message: "Server heartbeat succeeded"
+              awaited: { $$exists: true }
+              topologyId: { $$exists: true }
+              serverHost: { $$type: string }
+              serverPort: { $$type: [int, long] }
+              driverConnectionId: { $$exists: true }
+              serverConnectionId: { $$exists: true }
+              durationMS: { $$type: [int, long] }
+              reply:
+                $$matchAsDocument:
+                  "$$matchAsRoot":
+                    ok: 1
+  - description: Failing hearbeat
```

Comment:
```suggestion
  - description: Failing heartbeat
```

Reply:
Done, tnx.

Reply:
i think this one still needs to be fixed as well